src/main/java/gift/application/ProductService.java

```diff
@@ -0,0 +1,73 @@
+package gift.application;
+
+import gift.dao.ProductRepository;
+import gift.domain.Product;
+
+import gift.dto.ProductRequest;
+import gift.dto.ProductResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+
+    private final ProductRepository productRepository;
+
+    @Autowired
+    public ProductService(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
```

Comment:
[반영하지 않으셔도 좋습니다😀]

매개변수의 이름과 일치하는 빈 이름으로 주입할 수 있습니다. 예를 들어, 다음과 같이 매개변수 이름을 지정하면, 해당 이름과 일치하는 빈이 주입됩니다. MemoryProductRepository 에도 @Repository 를 추가하여 빈으로 사용되어야 하는 상황이라면 이렇게 bean 을 주입시킬 수 있을 것 같습니다.

```suggestion
    public ProductService(ProductRepository productDao) {
        this.productRepository = productDao;
    }
```